### PR TITLE
Added TryGetValue<T> for Properties - fix #21343

### LINKED ISSUE #21343: `EntityEntry.CurrentValues`
`EntityEntry.CurrentValues` throws an 'System.InvalidOperationException' when accessing a property which does not exist.

Just like `EntityEntry.CurrentValues.GetValue<T>`, I believe we should have:

- `+ public bool EntityEntry.CurrentValues.TryGetValue<T>(string propertyName, out T value)` which will indicate if the property existed or not without having to deal with the stack preservation overhead.

Perhaps the out should be a `PropertyValue` or there should also be a corresponding overload.

**Scenario:**
Consider implementing Soft Delete on some objects but not all or having `Created/ModifiedDate` or some combination therein on some objects but not all.

**Work around:**
Write trivial extension method, which unfortunately still have the overhead of the exception.

Using EntityFrameworkCore 3.1.5

### PR DESCRIPTION
**Pull request by m4ss1m0g**
Summary of the changes
- Added on abstract class `Microsoft.EntityFrameworkCore.ChangeTracking.PropertyValues` the method `public abstract bool TryGetValue<TValue>([NotNull] string propertyName, out TValue value);`
- The method implementation on derived classes
  ° `CurrentProertyValues`
  ° `ArrayPropertyValues`
  ° `OriginalPropertyValues`
- Added unit test `test/EFCore.Tests/ChangeTracking/Internal/PropertyValuesTest.cs`

Fixes #21343

### CODE REVIEW

File: src/EFCore/ChangeTracking/Internal/ArrayPropertyValues.cs [hunk 1]
```diff
@@ -153,6 +153,25 @@ public override IReadOnlyList<IProperty> Properties
         public override TValue GetValue<TValue>(string propertyName)
             => (TValue)this[propertyName];
 
+        /// <summary>
+        ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+        ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+        ///     any release. You should only use it directly in your code with extreme caution and knowing that
+        ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+        /// </summary>
+        public override bool TryGetValue<TValue>(string propertyName, out TValue value)
```
Comment (ajcvickers) [inline src/EFCore/ChangeTracking/Internal/ArrayPropertyValues.cs:162]: Does this need to be re-implemented in subclass overrides? It looks like maybe the implementation can be common and in the base class?

Comment (m4ss1m0g) [reply src/EFCore/ChangeTracking/Internal/ArrayPropertyValues.cs:162]: I haven't found a base, only abstract classes. I have implemented the method on concrete classes that implement the `PropertyValues` abstract class. This is my first PR on this project, maybe I missed something. Can you give me some clue ? Thanks.

Comment (m4ss1m0g) [reply src/EFCore/ChangeTracking/Internal/ArrayPropertyValues.cs:162]: @ajcvickers I understand you prev message, sorry I implement the code un PropertyValues class.

**Comment by ajcvickers** [comment]
@m4ss1m0g Looking better, but tests are failing. Take a look at [Getting and building the code](https://github.com/dotnet/efcore/blob/main/docs/getting-and-building-the-code.md) for how you can run all the tests locally.

File: src/EFCore/ChangeTracking/PropertyValues.cs [hunk 2]
```diff
@@ -148,6 +148,28 @@ public virtual IEntityType EntityType
...
+        /// <typeparam name="TValue"> The type of the property. </typeparam>
+        /// <param name="propertyName"> The property name. </param>
+        /// <param name="value"> The property value if any. </param>
+        /// <returns> True if the property exists, otherwise false. </returns>
+        public bool TryGetValue<TValue>([NotNull] string propertyName, out TValue value)
+        {
+            var property = EntityType.FindProperty(propertyName);
+            if (propertyName != null)
+            {
+                value = this.GetValue<TValue>(propertyName);
```
Comment (smitpatel) [inline src/EFCore/ChangeTracking/PropertyValues.cs:165]:
```suggestion
                value = GetValue<TValue>(propertyName);
```

**Comment by m4ss1m0g** [comment]
The line `var property = EntityType.FindProperty(propertyName);` raise the exception, I'm read properties through var property = Properties.FirstOrDefault(p=> p.Name == propertyName); I hope is ok.

**Comment by ajcvickers** [comment]
@m4ss1m0g Thanks for the contribution!

**Pull request merged**